### DISCUSSION TITLE
Crunching the National Water Model

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [nebari-dev/nebari#200] [Show and tell]
![2020-12-10_6-46-45](https://user-images.githubusercontent.com/1872600/101768446-93e8da00-3ab3-11eb-98c0-be9a1700587c.png)

The National Water Model Reanalysis v2.0 is a 26 year simulation of 2.7 million rivers in the US at hourly intervals.   The data was delivered as part of the NOAA Big Data Program to AWS as 227,000+ hourly NetCDF files.  

I downloaded (!) and then converted the streamflow files from the reanalysis to a single Zarr dataset with chunks that had a dimension of 100 in the time dimension to facilitate the extraction of time series data.  I used [rechunker](https://rechunker.readthedocs.io/en/latest/), and to deal with potential input data problems, I looped through the data in month-long chunks, writing and then appending to Zarr at the end of every month.  This way I could correct issues with the input data (missing data and bad time stamps), the try again, and on success, append the chunk. See the [full notebook](https://nbviewer.jupyter.org/gist/rsignell-usgs/c0b87ed1fa5fc694e665fb789e8381bb) for details on the conversion. 

The result is a single Zarr dataset on AWS that can be used for time series extraction as well as mapping.   

Here's proof:  A [sample analysis notebook](https://nbviewer.jupyter.org/gist/rsignell-usgs/78a4ce00360c65bc99764aa3e88a2493
) using this new Zarr dataset proof.  

In this notebook we use a cluster of 20 workers on a Qhub Dask Gateway cluster to both extract time series and compute the annual mean river discharge for a specific year in less than 2 minutes of wall clock. 

![2020-12-10_6-47-08](https://user-images.githubusercontent.com/1872600/101768471-9cd9ab80-3ab3-11eb-992d-426417679021.png)

**Comment by Adam-D-Lewis** [comment]
I'm closing old discussions in a repo cleanup effort.  Feel free to re-open if needed.
